recipes/mdspan/all/conanfile.py

```diff
@@ -0,0 +1,61 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MDSpanConan(ConanFile):
+    name = "mdspan"
+    homepage = "https://github.com/kokkos/mdspan"
+    description = "Production-quality reference implementation of mdspan"
+    topics = ("multi-dimensional", "array", "span")
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "compiler"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "3.4",
+            "apple-clang": "5.1"
+        }
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(
+            str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} "
+                             "compiler support.".format(
+                                 self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(
+                    "{} requires C++{} support. "
+                    "The current compiler {} {} does not support it.".format(
+                        self.name, self._minimum_cpp_standard,
+                        self.settings.compiler,
+                        self.settings.compiler.version))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "mdspan-mdspan-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
```

Comment:
This is a new feature and it helps remove some of the recent bugs about `os.rename` not working on some platforms (it uses the internal `tools.rename` from conan which is more robust)

```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

Reply:
Thanks! Good to know :-D